### ISSUE TITLE
[api-extractor] “The symbol React is being imported after it was already registered as non-imported” when running CLI

### ISSUE THREAD

**Issue by eelco** [Microsoft/web-build-tools#847]
When running API extractor on our library that uses React, it will throw the error:

> Error: Program Bug: The symbol React is being imported after it was already registered as non-imported

I’ve read the [source code](https://github.com/Microsoft/web-build-tools/blob/08f5c153b9459e8b4224244b12ef915e7108ce5a/apps/api-extractor/src/generators/dtsRollup/AstSymbolTable.ts#L408) that throws this error, but I’m using API extractor from the command line and have a hard time figuring out what would be causing this.

I *think* this might be a faulty assumption in the strategy used to detect external declarations.

**Comment by pgonzal** [comment]
Are you able to share a GitHub branch that repros the error?  That might help us to investigate?

FYI we recently upgraded API Extractor to use TypeScript 3.0 as its analysis engine.  When processing a project that compiles using TypeScript 2.0, you may need to use the [--typescript-lib-package](https://github.com/Microsoft/web-build-tools/pull/832) or [--skip-lib-check](https://github.com/Microsoft/web-build-tools/pull/844) compatibility options which allow the 3.0 engine to analyze 2.0 *.d.ts files.

**Comment by eelco** [comment]
The project uses TypeScript 3.0.1. I’ll try to make a minimal example to show the error, it’s not open source (yet).

**Comment by eelco** [comment]
Got it! Turns out there was a file in the project that *did* use React types, but did not import React. This works fine in VSCode and when compiling, but does trip up api-extractor.

The fix is adding `import * as React from "react"` at the top of the file.

---

I also found a related bug in trying to reduce the problem. If the project is one file with just
```
export type MyComponent = React.Component
```

That will cause `Error: Child declaration not found for the specified node`

Debugging this shows that the `node.symbol.name` is `ReactType`, which seems to indicate that the React types are being read to some level, but that it tries resolve the `React.Component` type in the AST instead of seeing it as external?
